In [ ]:
'''
client id = 
pic16bralphsproject-2432612430342441774c683551794d774d363142475571574978582e4f556f682e535843714f39634371486d513949536b36416e634850502e6b30578177261567869932466
client secret =
Q8DpUa-2BtFmCyABTf6NlNCXjsGrtpLOqjYBdWtf
'''

In [16]:
import requests
from requests.auth import HTTPBasicAuth

client_id = "pic16bralphsproject-2432612430342441774c683551794d774d363142475571574978582e4f556f682e535843714f39634371486d513949536b36416e634850502e6b30578177261567869932466"
client_secret = "Q8DpUa-2BtFmCyABTf6NlNCXjsGrtpLOqjYBdWtf"

auth_url = "https://api-ce.kroger.com/v1/connect/oauth2/token"

# Make a POST request to get the access token
response = requests.post(
    auth_url,
    auth=HTTPBasicAuth(client_id, client_secret),
    headers={"Content-Type": "application/x-www-form-urlencoded"},
    data={"grant_type": "client_credentials", "scope": "product.compact"}
)

# Check if the request was successful and extract the token
if response.status_code == 200:
    access_token = response.json().get("access_token")
    print("Access Token:", access_token)
else:
    print("Failed to get access token. Status Code:", response.status_code)
    print("Response Text:", response.text)


Access Token: eyJhbGciOiJSUzI1NiIsImprdSI6Imh0dHBzOi8vYXBpLWNlLmtyb2dlci5jb20vdjEvLndlbGwta25vd24vandrcy5qc29uIiwia2lkIjoidnl6bG52Y3dSUUZyRzZkWDBzU1pEQT09IiwidHlwIjoiSldUIn0.eyJhdWQiOiJwaWMxNmJyYWxwaHNwcm9qZWN0LTI0MzI2MTI0MzAzNDI0NDE3NzRjNjgzNTUxNzk0ZDc3NGQzNjMxNDI0NzU1NzE1NzQ5Nzg1ODJlNGY1NTZmNjgyZTUzNTg0MzcxNGYzOTYzNDM3MTQ4NmQ1MTM5NDk1MzZiMzY0MTZlNjM0ODUwNTAyZTZiMzA1NzgxNzcyNjE1Njc4Njk5MzI0NjYiLCJleHAiOjE3MzA3NTkwOTUsImlhdCI6MTczMDc1NzI5MCwiaXNzIjoiYXBpLWNlLmtyb2dlci5jb20iLCJzdWIiOiI4MzdhZWI4Zi01NjM3LTUxNTQtOGE1MC0wNjc5YjZhYzZjYmEiLCJzY29wZSI6InByb2R1Y3QuY29tcGFjdCIsImF1dGhBdCI6MTczMDc1NzI5NTg0NjE2MzI5NSwiYXpwIjoicGljMTZicmFscGhzcHJvamVjdC0yNDMyNjEyNDMwMzQyNDQxNzc0YzY4MzU1MTc5NGQ3NzRkMzYzMTQyNDc1NTcxNTc0OTc4NTgyZTRmNTU2ZjY4MmU1MzU4NDM3MTRmMzk2MzQzNzE0ODZkNTEzOTQ5NTM2YjM2NDE2ZTYzNDg1MDUwMmU2YjMwNTc4MTc3MjYxNTY3ODY5OTMyNDY2In0.qBnXB_bK4JWVGuK85mKKBSBswDRffYol-mdzSFFub17ym-arfGhnKx9aMcefdZfbO51pP83s-VIh4bFK-80a9K923quyIUVVZtiCFwmkNLdCCX5hIWwoZiUEE_xna3fsFakTNr5J3zK4yFgKQ3FYZLH8DB2Po5hBQo

In [17]:
headers = {
    "Authorization": f"Bearer {access_token}",
    "Accept": "application/json"
}
# Make the API request to get locations near the specified ZIP code
response = requests.get(
    "https://api-ce.kroger.com/v1/locations",
    headers=headers,
    params={"filter.zipCode.near": "90024", "filter.limit": 5}  # Replace with your ZIP code
)

# Check if the request was successful
if response.status_code == 200:
    # Parse the JSON response
    data = response.json().get("data", [])
    # Print each location's details
    for location in data:
        location_id = location.get("locationId")
        name = location.get("name")
        address = location.get("address", {})
        address_line1 = address.get("addressLine1")
        city = address.get("city")
        state = address.get("state")
        zip_code = address.get("zipCode")

        # Print out formatted details for each location
        print("\nLocation ID:", location_id)
        print("Store Name:", name)
        print("Address:", address_line1)
        print("City:", city)
        print("State:", state)
        print("ZIP Code:", zip_code)
        print("-" * 40)
else:
    print("Error:", response.status_code, response.json())



Location ID: 70300759
Store Name: Ralphs Fresh Fare - Westwood
Address: 10861 Weyburn Ave
City: Los Angeles
State: CA
ZIP Code: 90024
----------------------------------------

Location ID: 70300156
Store Name: Ralphs Fresh Fare - Century City
Address: 10309 W Olympic Blvd
City: Los Angeles
State: CA
ZIP Code: 90064
----------------------------------------

Location ID: 70300044
Store Name: Ralphs Fresh Fare - Bundy
Address: 12057 Wilshire Blvd
City: Los Angeles
State: CA
ZIP Code: 90025
----------------------------------------

Location ID: 70300210
Store Name: Ralphs Fresh Fare - Olympic Barrington
Address: 11727 W Olympic Blvd
City: Los Angeles
State: CA
ZIP Code: 90064
----------------------------------------

Location ID: 70300709
Store Name: Ralphs - Westdale S C
Address: 11361 National Blvd
City: Los Angeles
State: CA
ZIP Code: 90064
----------------------------------------


* Location ID: 70300759
* Store Name: Ralphs Fresh Fare - Westwood
* Address: 10861 Weyburn Ave
* City: Los Angeles
* State: CA
* ZIP Code: 90024

In [30]:
# Make the API request
headers = {
    "Authorization": f"Bearer {access_token}",
    "Accept": "application/json"
}
#, "filter.locationId": "70300759"
params ={"filter.term":"butter"}

response = requests.get("https://api-ce.kroger.com/v1/products", headers=headers, params=params)

# Check if the request was successful
if response.status_code == 200:
    try:
        data = response.json().get("data", [])
        for product in data:
            print("Product Name:", product.get("description"))
            print("Brand:", product.get("brand"))
            print("Category:", product.get("categories", [None])[0])
            print("Size:", product.get("items", [{}])[0].get("size"))
            print("Price:", product.get("items", [{}])[0].get("price", {}).get("regular"))
            print("-" * 40)
    except requests.JSONDecodeError:
        print("Error parsing JSON response.")
else:
    # Print status code and raw text if the response is not JSON
    print("Request failed.")
    print("Status Code:", response.status_code)
    print("Response Text:", response.text)


Product Name: Kroger® Spreadable Butter with Canola Oil
Brand: Kroger
Category: Dairy
Size: 8 oz
Price: None
----------------------------------------
Product Name: Country Crock Dairy Free Olive Oil Plant Butter
Brand: Country Crock
Category: Dairy
Size: 16 oz
Price: None
----------------------------------------
Product Name: Land O’ Lakes Canola Oil Light Butter Spread
Brand: Land O' Lakes
Category: Dairy
Size: 8 oz
Price: None
----------------------------------------
Product Name: Simple Truth Organic™ Salted Butter
Brand: Simple Truth Organic
Category: Natural & Organic
Size: 16 oz
Price: None
----------------------------------------
